server/py/framework/api/utils.py

```diff
@@ -204,11 +206,87 @@ def _generate_function_and_task_from_submit_run_body(db_session: Session, data):
 
 
 async def submit_run(
-    db_session: Session, auth_info: mlrun.common.schemas.AuthInfo, data
+    db_session: Session,
+    auth_info: mlrun.common.schemas.AuthInfo,
+    background_tasks: fastapi.BackgroundTasks,
+    data,
 ):
-    _, _, _, response = await run_in_threadpool(
-        submit_run_sync, db_session, auth_info, data
+    from services.api.utils.endpoints import (
+        start_model_endpoint_creation_background_task,
     )
+
+    response = None
+
+    try:
+        fn, task = _generate_function_and_task_from_submit_run_body(db_session, data)
+        run_db = get_run_db_instance(db_session)
+        fn.set_db_connection(run_db)
+
+        track_models = getattr(fn.spec, "track_models", False)
+        logger.info(
+            "Starting model endpoint creation?",
+            track_models=track_models,
+            background_tasks=str(background_tasks),
+            db_session=str(db_session),
+        )
+
+        if track_models and background_tasks and db_session:
+            project = task["metadata"]["project"]
+            function_name = fn.metadata.name
+            (
+                fn,
+                model_endpoint_creation_task_name,
+                _,
+            ) = await start_model_endpoint_creation_background_task(
+                project=project,
+                name=function_name,
+                background_tasks=background_tasks,
+                function=fn.to_dict(),
+                db_session=db_session,
+                is_batch=True,
+            )
+            fn = mlrun.new_function(
+                runtime=fn,
+                project=project,
+                name=function_name,
+            )
+            fn.spec.model_endpoint_creation_task_name = (
+                model_endpoint_creation_task_name
+            )
+
+            # TODO: there should be a better way to do this
+            serving_spec = getattr(fn.spec, "serving_spec")
+            if serving_spec:
+                serving_spec = json.loads(serving_spec)
+                serving_spec["model_endpoint_creation_task_name"] = (
+                    model_endpoint_creation_task_name
+                )
```

Comment:
Why do we need the `model_endpoint_creation_task_name` in both the runtime and the serving spec?
It would be great if we could remove it from the general runtime spec, since it's not relevant for most runtimes.

Reply:
Not sure I follow. Why is `model_endpoint_creation_task_name` relevant to a "standard" nuclio function? It is only in use when deploying serving graphs. That's exactly why I don't understand why we need it in any other place other than the serving spec.

Reply:
Either way though, this isn't a change in this PR – it was introduced in #7150. Let's discuss it when you're back.